internal/provider/provider.go

```diff
@@ -205,7 +267,7 @@ func azureProvider(supportLegacyTestSuite bool) *schema.Provider {
 			"partner_id": {
 				Type:         schema.TypeString,
 				Optional:     true,
-				ValidateFunc: validation.Any(validation.IsUUID, validation.StringIsEmpty),
+				ValidateFunc: ValidatePartnerID,
```

Comment:
could we revert this change? using .Any makes it clear this can be either a partnerID or emty
```suggestion
				ValidateFunc: validation.Any(ValidatePartnerID, validation.StringIsEmpty),
```